### PR TITLE
475 error when adding new question page

### PR DESCRIPTION
**Pull request by pricea1**
### What is the context of this PR?
'Add Question Page' button at bottom of design page produced error. New tests added to cover Add Question Page button click.

### How to review 
1. On Master, Create new questionnaire and click 'Add Question Page' button beneath form design. Error occurs
2. On PR branch, repeat above step
3. Tests pass

### CODE REVIEW

File: eq-author/src/App/questionPage/Design/index.test.js [hunk 1]
```diff
@@ -438,10 +438,32 @@ describe("QuestionPageRoute", () => {
...
+          loading: false,
+          match,
+          page,
+          ...mockHandlers,
+        },
+        mount
+      );
+
+      wrapper
+        .find(`[data-test="btn-add-page"]`)
```
[the remark this answers is not in the document]
Comment (pricea1) [reply eq-author/src/App/questionPage/Design/index.test.js]: byTestAttr selector added for btn-add-page button

**Comment by pricea1** [comment]
[Error when adding new question page](https://trello.com/c/eSywymhK/1015-error-when-adding-new-question-page)

File: eq-author/src/App/questionPage/Design/index.test.js [hunk 2]
```diff
@@ -438,10 +439,32 @@ describe("QuestionPageRoute", () => {
         .simulate("click");
 
       expect(mockHandlers.onDuplicatePage).toHaveBeenCalledWith({
-        sectionId: match.params.sectionId,
+        sectionId: page.section.id,
```
Comment (tgandrews) [inline eq-author/src/App/questionPage/Design/index.test.js]: How is this test not failing? Are they both `undefined`?

File: eq-author/cypress/integration/authenticated/builder_spec.js [hunk 3]
```diff
@@ -510,6 +510,16 @@ describe("builder", () => {
     });
   });
 
+  it("Can add a new question page", () => {
```
Comment (SamGodwin2) [inline eq-author/cypress/integration/authenticated/builder_spec.js]: Can we move this test into the other add page test and explain in that test title that we have two different methods of adding a page?

File: eq-author/src/App/questionPage/Design/index.test.js (hunk 2 above)
Comment (pricea1) [reply eq-author/src/App/questionPage/Design/index.test.js]: Good question. I added in an extra test to see what was happening:

Expected mock function to have been called with:
      {"ignoreMe": 1, "pageId": undefined, "position": 1, "sectionId": undefined}

 as argument 1, but it was called with
      {"pageId": undefined, "position": 1}.

Looks like Jest is ignoring undefined parameters.

I've removed sectionId from test for clarity

File: eq-author/cypress/integration/authenticated/builder_spec.js (hunk 3 above)
Comment (pricea1) [reply eq-author/cypress/integration/authenticated/builder_spec.js]: I've moved the new test into the existing add page test

**Comment by pricea1** [comment]
Fixes https://github.com/ONSdigital/eq-author-app/issues/282

**Pull request merged**